src/coreclr/jit/compiler.hpp

```diff
@@ -1489,13 +1489,7 @@ inline GenTreeIntCon* Compiler::gtNewIconHandleNode(size_t value, GenTreeFlags f
 
 inline GenTree* Compiler::gtNewIconEmbScpHndNode(CORINFO_MODULE_HANDLE scpHnd)
 {
-    void *embedScpHnd, *pEmbedScpHnd;
-
-    embedScpHnd = (void*)info.compCompHnd->embedModuleHandle(scpHnd, &pEmbedScpHnd);
-
-    assert((!embedScpHnd) != (!pEmbedScpHnd));
-
-    return gtNewIconEmbHndNode(embedScpHnd, pEmbedScpHnd, GTF_ICON_SCOPE_HDL, scpHnd);
+    return gtNewIconEmbHndNode((void*)scpHnd, nullptr, GTF_ICON_SCOPE_HDL, scpHnd);
```

Comment:
Isn't this introducing an abstraction violation? `CORINFO_XXX_HANDLE`s are not meant to be embedded in the code. We have an intentional distinction between the `CORINFO_XXX_HANDLE` handles used to represent type system entities on JIT/EE interface and how these values get embedded into the code.

It happens to work since these module handles are only used with JIT currently, but this change does not look like an improvement to me.

Reply:
@jkotas I just assumed that almost 300 LOC is too much for an API that is just no-op (and CoreCLR) only, I can revert this part.

Looks like the only case where we need to embed a field handle is CoreCLR jitting a method that accesses a field it has no access for, so JIT emits an exception telling about the field access violation. Isn't part of the IL verification that we got rid some time ago?